### PR TITLE
add unit tests for azuremachinetemplate_webhook ValidateCreate

### LINKED ISSUE #3429: Add missing test coverage for azuremachinetemplate_webhook.go
/kind cleanup

**Describe the solution you'd like**
Currently, `azuremachinetemplate_webhook.go` is missing unit tests in the following areas:

- [ValidateCreate()](https://app.codecov.io/github/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/azuremachinetemplate_webhook.go#L56) needs to be tested for the following cases:
  - spec.RoleAssignmentName != ""
  - (r.Spec.Template.Spec.NetworkInterfaces != nil) && len(r.Spec.Template.Spec.NetworkInterfaces) > 0 && r.Spec.Template.Spec.SubnetName != ""
  - (r.Spec.Template.Spec.NetworkInterfaces != nil) && len(r.Spec.Template.Spec.NetworkInterfaces) > 0 && r.Spec.Template.Spec.AcceleratedNetworking != nil
  - networkInterface.PrivateIPConfigs < 1

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by samyakjain10**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3429 

**Special notes for your reviewer**:

**TODOs**:

- [X] squashed commits
- [X] adds unit tests

**Release note**:

```release-note
None
```

### CODE REVIEW

File: api/v1beta1/azuremachinetemplate_webhook_test.go [hunk 1]
```diff
@@ -128,15 +128,109 @@ func TestAzureMachineTemplate_ValidateCreate(t *testing.T) {
...
+			name:            "azuremachinetemplate without SystemAssignedIdentityRoleName",
+			machineTemplate: createAzureMachineTemplateFromMachine(createMachineWithoutSystemAssignedIdentityRoleName()),
+			wantErr:         false,
+		},
 		{
 			name:            "azuremachinetemplate without RoleAssignmentName",
 			machineTemplate: createAzureMachineTemplateFromMachine(createMachineWithoutRoleAssignmentName()),
 			wantErr:         false,
 		},
+		{
```
Comment (willie-yao) [inline api/v1beta1/azuremachinetemplate_webhook_test.go:145]: nit: Can you move this test above the one without RoleAssignmentName just for consistency?

**Review by willie-yao — COMMENTED** [review]
Awesome work! Just a minor nit and this looks good to me.

File: api/v1beta1/azuremachinetemplate_webhook_test.go (hunk 1 above)
Comment (samyakjain10) [reply api/v1beta1/azuremachinetemplate_webhook_test.go]: That makes sense! Made the change

**Review by willie-yao — APPROVED** [review]
/lgtm

Great work @samyakjain10 🚀

**Review by mboersma — APPROVED** [review]
/lgtm
/approve

**Pull request merged**